.github/scripts/js/ci.js

```diff
@@ -79,116 +79,262 @@ module.exports.updateCommentOnStart = async ({ github, context, core, name }) =>
+    statusComment = `:white_circle:\u00a0\`${name}\` for \`${ref}\` [skipped](${build_url}).`;
+  }
+
+  return statusComment;
+}
+
+/**
+ * Update a comment in "release" issue or pull request with status of the job or the workflow.
+ *
+ * statusConfig values:
+ * "job" - get status from job context
+ * "workflow" - calculate workflow status from needs context
+ * "one-line" - Report job status as one line to form one huge multiline.
+ * "separate" - Report job statuses on separate lines.
+ * "no-skipped" - do not report skipped and cancelled jobs
+ * "final" - restore statuses from needs context, wrap comment with details and add summary status for the workflow.
+ * "restore-separate" - restore job statuses as separate lines
+ * "restore-one-line" - restore job statuses as one-line.
+ *
+ * Examples:
  * "job,inline" updates comment with job status and a name without extra newlines.
  * "job" updates comment with job status and a name with extra newlines.
- * "workflow" updates comment with statuses of all jobs in needs context with extra newlines.
+ * "workflow,final,no-skipped" add statuses of all jobs without skipped and canceled.
  *
  * @param {object} inputs
  * @param {object} inputs.github - A pre-authenticated octokit/rest.js client with pagination plugins.
  * @param {object} inputs.context - An object containing the context of the workflow run.
  * @param {object} inputs.core - A reference to the '@actions/core' package.
- * @param {string} inputs.statusSource - 'job,inline', 'job' or 'workflow'.
+ * @param {string} inputs.statusConfig - A comma-separated combination of 'job', 'workflow', 'one-line', 'separate', 'no-skipped' and 'final'.
  * @param {string} inputs.name - A name to use in the comment.
  * @param {object} inputs.needsContext - The needs context contains outputs from all jobs that are defined as a dependency of the current job.
  * @param {object} inputs.jobContext - The job context contains information about the currently running job.
+ * @param {object} inputs.stepsContext - The steps context contains information about previously executed steps.
+ * @param {object} inputs.jobNames - An object with each job names.
  * @returns {Promise<*>}
  */
-module.exports.updateCommentOnFinish = async ({ github, context, core, statusSource, name, needsContext, jobContext }) => {
+module.exports.updateCommentOnFinish = async ({ github, context, core, statusConfig, name, needsContext, jobContext, stepsContext, jobNames }) => {
+  const repo_url = context.payload.repository.html_url;
+  const run_id = context.runId;
+  const build_url = `${repo_url}/actions/runs/${run_id}`;
+  const ref = context.payload.inputs.pull_request_head_label || context.ref;
+
+  // Get elapsed time
+  let time_elapsed = null;
+  let startedAt = null;
+  if (statusConfig.includes('job') && stepsContext['started_at'] && stepsContext['started_at'].outputs['started_at']) {
+    // Calculate workflow elapsed time using 'started_at' step.
```

Comment:
If we could use Node 14, there optional chaining would available for us

```
$ node
Welcome to Node.js v14.19.0.
Type ".help" for more information.
> const obj = {a:{b: {c : 3 }}}
undefined
> obj?.a?.b?.c
3
> obj?.XXX?.b?.c
undefined
> 
```

Reply:
Github-script 6 is released! Node 16 is available now.

Reply:
It seems a separate task: migrate to github-script 6.